scripts/opportunity-provider-limo/index.ts

```diff
@@ -0,0 +1,206 @@
+    if (isNaN(price)) {
+      throw new Error(`Invalid price: ${data.price}`);
+    }
+    prices[index] = price;
+  }
+
+  const mintDecimals = await getDecimals(connection, token);
+  return prices[index] / Math.pow(10, mintDecimals);
+}
+
+async function createOpportunities(
+  skExecutor: Keypair,
+  limoClient: limo.LimoClient,
+  opportunitiesPath: string,
+  count: number,
+  edge: number
+) {
+  const opportunities = loadOpportunities(opportunitiesPath);
+  for (let i = 0; i < opportunities.length; i++) {
+    const opportunity = opportunities[i];
+    for (let j = 0; j < count; j++) {
+      let inputToken = opportunity.token1;
+      let outputToken = opportunity.token2;
+      if (opportunity.randomizeSides) {
+        if (Math.random() > 0.5) {
+          inputToken = opportunity.token2;
+          outputToken = opportunity.token1;
+        }
+      }
+
+      const priceInput = await getPrice(limoClient.getConnection(), inputToken);
+      const priceOutput = await getPrice(
+        limoClient.getConnection(),
+        outputToken
+      );
+
+      const notional =
+        Math.random() *
+          (opportunity.maxAmountNotional - opportunity.minAmountNotional) +
+        opportunity.minAmountNotional;
+
+      const amountInput = (notional * (1 + edge / 10_000)) / priceInput;
+      const amountOutput = notional / priceOutput;
+
+      console.log("Creating opportunity:");
+      const decimalsInput = await getDecimals(
+        limoClient.getConnection(),
+        inputToken
+      );
+      const decimalsOutput = await getDecimals(
+        limoClient.getConnection(),
+        outputToken
+      );
+      console.log(
+        `Input: ${inputToken.symbol}, ${
+          amountInput / Math.pow(10, decimalsInput)
+        }`
+      );
+      console.log(
+        `Output: ${outputToken.symbol}, ${
+          amountOutput / Math.pow(10, decimalsOutput)
+        }`
+      );
+
+      const signature = (
+        await limoClient.createOrderGeneric(
+          skExecutor,
+          inputToken.mint,
+          outputToken.mint,
+          new Decimal(amountInput.toFixed()),
+          new Decimal(amountOutput.toFixed())
+        )
+      )[0];
+      console.log(`Created opportunity: ${signature}`);
+    }
+  }
+}
+
+const argv = yargs(hideBin(process.argv))
+  .option("sk-payer", {
+    description:
+      "Secret key of address to submit transactions with. If action is 'create', this keypair creates the order in Limo. In 64-byte base58 format",
+    type: "string",
+    demandOption: true,
+  })
+  .option("global-config", {
+    description: "Global config address",
+    type: "string",
+    demandOption: true,
+  })
+  .option("endpoint-svm", {
+    description: "SVM RPC endpoint",
+    type: "string",
+    demandOption: true,
+  })
+  .option("opportunities", {
+    description: "Path to opportunities file",
+    type: "string",
+    default: "opportunities.json",
+  })
+  .option("count", {
```

Comment:
if we want to run this in a loop, we should adjust the code. Otherwise, the deployment will be in some sort of crash loop